### DISCUSSION TITLE
Custom LookupScript

### DISCUSSION THREAD

**Discussion by mgdm187** [serenity-is/Serenity#5520] [Help]
Hi everyone, 
I migrated my project from .net to .net5 and I am getting some errors in my custom lookup script. 
`[LookupScript]
    public class MyCustomerLookup : RowLookupScript<Entities.CustomerRow>
    {
        public MyCustomerLookup()
            : base()
        {
            IdField = TextField = "Customer";
        }

        protected override void PrepareQuery(SqlQuery query)
        {
            var fld = Entities.CustomerRow.Fields;
            query.Distinct(true)
                .Select(fld.Customer)
                .Where(
                    new Criteria(fld.Customer) != "" &
                    new Criteria(fld.Customer).IsNotNull());
        }

        protected override void ApplyOrder(SqlQuery query)
        {
        }
    }`
This is the code for my custom lookup script but I get an error saying: There is no argument given that corresponds to the required formal parameter 'connections' of 'RowLookupScript<CustomerRow>.RowLookupScript(ISqlConnections)'.  The base() word is underlined and this is the code for RowLookupScript:
`public class RowLookupScript<TRow> : LookupScript where TRow : class, IRow, new()
    {
        public RowLookupScript(ISqlConnections connections);

        protected virtual void ApplyOrder(SqlQuery query);
        protected override IEnumerable GetItems();
        protected virtual void PrepareQuery(SqlQuery query);
    }`
I am missing that SQL connection and not sure how to use it.
Thank you in advance.

**Comment by luongnv6** [comment]
It's not recommended to use base () in this case

**Reply by juliermefelix** [reply]
use(ISqlConnections connections)
             : base(connections)

example:
  public TurmaLookup(ISqlConnections connections)
             : base(connections)
         {
             IdField = TextField = TurmaRow.Fields.Turma.PropertyName;
         }
